src/scanner.cc

```diff
@@ -73,6 +74,11 @@ struct Scanner {
       lexer->advance(lexer, true);
     }
 
+    if (lexer->lookahead == 0) {
+      lexer->result_symbol = EOF_NEWLINE;
+      return true;
+    }
```

Comment:
@maxbrunsfeld I tried at first to use `NEWLINE`, but I'm not able to configure the grammar such that infinite looping doesn't occur, or that we still pass all tests and edge cases. From trying to find a way to make this work with only `NEWLINE`, I've found having a separate token indicating EOF makes this easier to reason about. Let me know what you think, or if you'd like to take a look at this together! :D